mypyc/ops_misc.py

```diff
@@ -379,6 +379,12 @@
     error_kind=ERR_NEVER,
     emit=call_emit('PyObject_Type'))
 
+type_object_op = custom_op(result_type=object_rprimitive,
```

Comment:
Make this a `name_ref_op` (like `ellipsis_op`, among others), so then it will get picked up by the primitive matcher as a nice side effect (which will generate better code in some places that refer to `type`)